packages/kit/src/views/Swap/hooks/useSwapTxHistory.ts

```diff
@@ -0,0 +1,354 @@
+      );
+      setSwapHistory(sortHistories);
+    },
+    [setSwapHistory],
+    {
+      watchLoading: true,
+    },
+  );
+  return { syncLoading: isLoading };
+}
+
+export function useSwapTxHistoryStateSyncInterval() {
+  const [swapTxHistoryPending] = useSwapTxHistoryPendingAtom();
+  const { updateSwapHistoryItem } = useSwapActions().current;
+  const internalRef = useRef<Record<string, NodeJS.Timeout>>({});
+  const swapTxHistoryPendingRef = useRef<ISwapTxHistory[]>([]);
+  if (swapTxHistoryPendingRef.current !== swapTxHistoryPending) {
+    swapTxHistoryPendingRef.current = swapTxHistoryPending;
+  }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const triggerSwapPendingHistoryInterval = useCallback(
+    debounce(() => {
+      if (swapTxHistoryPendingRef.current.length > 0) {
+        swapTxHistoryPendingRef.current.forEach(async (swapTxHistory) => {
+          if (!internalRef.current[swapTxHistory.txInfo.txId]) {
+            const interval = setInterval(async () => {
+              const txStatusRes =
+                await backgroundApiProxy.serviceSwap.fetchTxState({
+                  txId: swapTxHistory.txInfo.txId,
+                  provider: swapTxHistory.swapInfo.provider.provider,
+                  protocol: EExchangeProtocol.SWAP,
+                  networkId: swapTxHistory.baseInfo.fromToken.networkId,
+                  ctx: swapTxHistory.ctx,
+                  toTokenAddress:
+                    swapTxHistory.baseInfo.toToken.contractAddress,
+                  receivedAddress: swapTxHistory.txInfo.receiver,
+                });
+              if (txStatusRes.state !== ESwapTxHistoryStatus.PENDING) {
+                clearInterval(interval);
+                delete internalRef.current[swapTxHistory.txInfo.txId];
+                await updateSwapHistoryItem({
+                  ...swapTxHistory,
+                  status: txStatusRes.state,
+                  txInfo: {
+                    ...swapTxHistory.txInfo,
+                    receiverTransactionId:
+                      txStatusRes.crossChainReceiveTxHash || '',
+                    gasFeeInNative: txStatusRes.gasFee
+                      ? txStatusRes.gasFee
+                      : swapTxHistory.txInfo.gasFeeInNative,
+                    gasFeeFiatValue: txStatusRes.gasFeeFiatValue
+                      ? txStatusRes.gasFeeFiatValue
+                      : swapTxHistory.txInfo.gasFeeFiatValue,
+                  },
+                  baseInfo: {
+                    ...swapTxHistory.baseInfo,
+                    toAmount: txStatusRes.dealReceiveAmount
+                      ? txStatusRes.dealReceiveAmount
+                      : swapTxHistory.baseInfo.toAmount,
+                  },
+                });
+              }
+            }, 1000 * 5);
+            internalRef.current[swapTxHistory.txInfo.txId] = interval;
+          }
+        });
+      }
+    }, 100),
+    [updateSwapHistoryItem],
+  );
```

Comment:
这段代码这样写嵌套很深，可以优化返回结构。forEach、setInterval 之间没有空隙很难阅读，也可以优化下，类似下面这样



```suggestion
  const triggerSwapPendingHistoryInterval = useCallback(
    debounce(() => {
    if (swapTxHistoryPendingRef.current.length === 0) {
      return;
    }

    swapTxHistoryPendingRef.current.forEach((swapTxHistory) => {
      if (internalRef.current[swapTxHistory.txInfo.txId]) {
        return;
      }

      const interval = setInterval(async () => {
        const txStatusRes = await backgroundApiProxy.serviceSwap.fetchTxState({ 
          // params
        });

        if (txStatusRes.state === ESwapTxHistoryStatus.PENDING) {
          return;
        }

        clearInterval(interval);
        delete internalRef.current[swapTxHistory.txInfo.txId];
        await updateSwapHistoryItem({ 
          // params
        });
      }, 1000 * 5);
      internalRef.current[swapTxHistory.txInfo.txId] = interval;
    });
  }, 100),
    [updateSwapHistoryItem],
  );
```

Reply:
1